src/app/code/community/Omise/Gateway/controllers/Callback/ValidateoffsiteinstalmentController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+class Omise_Gateway_Callback_ValidateoffsiteinstalmentController extends Omise_Gateway_Controller_Base {
+
+    public function indexAction() {
+        // Callback validation.
+        $order = $this->_getOrder();
+
+        if (!$payment = $order->getPayment()) {
+            Mage::getSingleton('core/session')->addError(
+                $this->__('Instalment payment validation failed, we cannot retrieve your payment information. Please contact our support team to confirm the payment.')
+            );
+
+            $this->_redirect('checkout/cart');
+            return;
+        }
+
+        $charge = Mage::getModel('omise_gateway/api_charge')->find(
+            $payment->getMethodInstance()->getInfoInstance()->getAdditionalInformation('omise_charge_id')
+        );
+
+        if ($charge instanceof Omise_Gateway_Model_Api_Error) {
+            Mage::getSingleton('core/session')->addError($charge->getMessage());
+            return $this->_redirect('checkout/cart');
+        }
+
+        if ($charge->isAwaitPayment()) {
+            $order->markAsAwaitPayment(
+                $payment->getLastTransId(),
+                Mage_Sales_Model_Order::STATE_PAYMENT_REVIEW,
+                Mage::helper('omise_gateway')->__('The payment is in progress.<br/>More explanation here...')
```

Comment:
@jonrandy `More explanation here` should be replaced with an actual message?

Reply:
I'm awaiting a solution for the 'cancelled' situation to decide what to put here. I'm still of the opinion we shouldn't release until this is resolved